.github/workflows/sub_gradle_test_and_build.yml

```diff
@@ -51,7 +54,18 @@ jobs:
       - name: Gradle test and build. (unit tests, integration tests, end-2-end tests and build)
         env:
           run_docker: false
-        run: ./gradlew clean build --no-daemon --stacktrace -x spotlessApply -x spotlessKotlinApply -x javadoc -x javadocJar -x sourcesJar
+        run: ./gradlew clean build jacocoTestReport --no-daemon --stacktrace -x spotlessApply -x spotlessKotlinApply -x javadoc -x javadocJar -x sourcesJar
+
+      - name: Add coverage to PR
+        id: jacoco
+        uses: madrapps/jacoco-report@v1.6.1
+        with:
+          paths: ${{ github.workspace }}/**/build/reports/jacoco/test/jacocoTestReport.xml
+          token: ${{ secrets.GITHUB_TOKEN }}
+          min-coverage-overall: 40
+          min-coverage-changed-files: 60
```

Comment:
Can this check be moved into the gradle build instead? That way, the developer can get feedback locally without publishing a PR.

Reply:
Also, do we want to set coverage minimums by project? I think it would make sense to have the `core` project require higher coverage than the reference server, for example.

Reply:
> Can this check be moved into the gradle build instead? That way, the developer can get feedback locally without publishing a PR.

Yes The JacocoCoverageVerification task can be used to verify if code coverage metrics are met based on configured rules. However the build fails if any of the configured rules are not met and JaCoCo only reports the first violated rule. I will gradually implemented in the following PR.

Right now you can simply run jacocoTestReport in local to get an report but without the delta.

> Also, do we want to set coverage minimums by project? I think it would make sense to have the `core` project require higher coverage than the reference server, for example.

I think that's part of the first question where we want enforce more granular test coverage rules.

Reply:
> Yes The JacocoCoverageVerification task can be used to verify if code coverage metrics are met based on configured rules. However the build fails if any of the configured rules are not met and JaCoCo only reports the first violated rule. I will gradually implemented in the following PR.

So once we enable `JacocoCoverageVerification`, do we still want to set `min-coverage-overall` and `min-coverage-changed-files` in the Github action?